docs/flask.asciidoc

```diff
@@ -0,0 +1,232 @@
+----
+
+[float]
+[[flask-debug-mode]]
+==== Debug Mode
+
+Please note that errors and transactions will only be sent to the apm server
+if your app is *not* in
+http://flask.pocoo.org/docs/0.12/quickstart/#debug-mode[debug mode].
+
+To force the agent to send data during development,
+set the value of `DEBUG` in the `ELASTIC_APM` dictionary to `True`:
+
+[source,python]
+----
+app.config['ELASTIC_APM'] = {
+        'APP_NAME': '<APP-NAME>',
+        'SECRET_TOKEN': '<SECRET-TOKEN>',
+        'DEBUG': True
+}
+----
+
+[float]
+[[flask-building-applications-on-the-fly]]
+==== Building applications on the fly?
+
+You can use the agent's `init_app` hook for adding the application on the fly:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+apm = ElasticAPM()
+
+def create_app():
+    app = Flask(__name__)
+    apm.init_app(app, app_name='<APP-NAME>', secret_token='<SECRET-TOKEN>')
+    return app
+----
+
+[float]
+[[flask-usage]]
+=== Usage
+
+Once you have configured the agent,
+it will automatically track transactions and capture uncaught exceptions within Flask.
+If you want to send additional events,
+a couple of shortcuts are provided on the ElasticAPM Flask middleware object
+by raising an exception or logging a generic message.
+
+Capture an arbitrary exception by calling `capture_exception`:
+
+[source,python]
+----
+try:
+    1 / 0
+except ZeroDivisionError:
+    apm.capture_exception()
+----
+
+Log a generic message with `capture_message`:
+
+[source,python]
+----
+apm.capture_message('hello, world!')
+----
+
+[float]
+[[flask-logging]]
+=== Logging
+
+Passing `logging=True` to the `ElasticAPM` constructor will make the agent
+automatically log errors that are logging using Python's built-in `logging` module.
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+apm = ElasticAPM(app, logging=True)
+----
+
+For fine-grained control, you can initialize a logging handler and add it,
+just as you would any other handler.
+
+[source,python]
+----
+from flask import Flask
+
+from elasticapm.contrib.flask import ElasticAPM
+from elasticapm.handlers.logging import LoggingHandler
+
+app = Flask(__name__)
+
+apm = ElasticAPM(app)
+
+if __name__ == '__main__':
+    # Create a logging handler and attach it.
+    handler = LoggingHandler(client=apm.client)
+    handler.setLevel(logging.WARN)
+    app.logger.addHandler(handler)
+----
+
+You can also capture exceptions and send them explictly:
+
+[source,python]
+----
+@app.route('/')
+def bar():
+    try:
+        1 / 0
+    except ZeroDivisionError:
+        app.logger.error( 'I cannot math', exc_info=True)
+----
+
+NOTE: `exc_info=True` adds the exception info to the data that gets sent to the APM server.
+Without it, only the message is sent.
+
+[float]
+[[flask-extra-data]]
+==== Extra data
+
+If you want to send more data to ElasticAPM than what you get with the agent by default,
+logging can be done like so:
```

Comment:
"In addition to what the agents log by default, you can send _extra_ information":